test/functional/feature_backwards_compatibility.py

```diff
@@ -156,173 +150,72 @@ def run_test(self):
         assert info['private_keys_enabled']
         assert info['keypoolsize'] == 0
 
-        # w3_v19: blank wallet, created with v0.19
-        node_v19.rpc.createwallet(wallet_name="w3_v19", blank=True)
-        wallet = node_v19.get_wallet_rpc("w3_v19")
-        info = wallet.getwalletinfo()
-        assert info['private_keys_enabled']
-        assert info['keypoolsize'] == 0
-
-        # w3_v18: blank wallet, created with v0.18
-        node_v18.rpc.createwallet(wallet_name="w3_v18", blank=True)
-        wallet = node_v18.get_wallet_rpc("w3_v18")
-        info = wallet.getwalletinfo()
-        assert info['private_keys_enabled']
-        assert info['keypoolsize'] == 0
-
         # Unload wallets and copy to older nodes:
         node_master_wallets_dir = os.path.join(node_master.datadir, "regtest/wallets")
         node_v19_wallets_dir = os.path.join(node_v19.datadir, "regtest/wallets")
-        node_v18_wallets_dir = os.path.join(node_v18.datadir, "regtest/wallets")
         node_v17_wallets_dir = os.path.join(node_v17.datadir, "regtest/wallets")
         node_v16_wallets_dir = os.path.join(node_v16.datadir, "regtest")
         node_master.unloadwallet("w1")
         node_master.unloadwallet("w2")
         node_master.unloadwallet("w3")
-        node_v19.unloadwallet("w1_v19")
-        node_v19.unloadwallet("w2_v19")
-        node_v19.unloadwallet("w3_v19")
-        node_v18.unloadwallet("w1_v18")
-        node_v18.unloadwallet("w2_v18")
-        node_v18.unloadwallet("w3_v18")
-
-        # Copy wallets to v0.16
-        for wallet in os.listdir(node_master_wallets_dir):
-            shutil.copytree(
-                os.path.join(node_master_wallets_dir, wallet),
-                os.path.join(node_v16_wallets_dir, wallet)
-            )
-
-        # Copy wallets to v0.17
-        for wallet in os.listdir(node_master_wallets_dir):
-            shutil.copytree(
-                os.path.join(node_master_wallets_dir, wallet),
-                os.path.join(node_v17_wallets_dir, wallet)
-            )
-        for wallet in os.listdir(node_v18_wallets_dir):
-            shutil.copytree(
-                os.path.join(node_v18_wallets_dir, wallet),
-                os.path.join(node_v17_wallets_dir, wallet)
-            )
 
-        # Copy wallets to v0.18
-        for wallet in os.listdir(node_master_wallets_dir):
-            shutil.copytree(
-                os.path.join(node_master_wallets_dir, wallet),
-                os.path.join(node_v18_wallets_dir, wallet)
-            )
-
-        # Copy wallets to v0.19
-        for wallet in os.listdir(node_master_wallets_dir):
-            shutil.copytree(
-                os.path.join(node_master_wallets_dir, wallet),
-                os.path.join(node_v19_wallets_dir, wallet)
-            )
+        for node in self.nodes[2:]:
```

Comment:
In commit "test: v0.20.1 backwards compatibility" (94d3989352aba63c86c901269438bfd566dd8632)

The deduplication here is great but the `nodes[2:]` references are more obscure and maybe fragile. Could this use `for node in legacy_nodes`? And

legacy_nodes = [node_v19, node_v18, node_v17]
or
legacy_nodes = nodes[2:]
or
legacy_nodes = [node for node in nodes if node.version <= 190000]

?

Reply:
That's a good idea...